avocado/core/test.py

```diff
@@ -397,9 +397,17 @@ def record_and_warn(*args, **kwargs):
         self.__workdir = None
         self.__srcdir = None
 
+        self.log.debug("Test metadata:")
         if self.filename:
-            self.log.debug("Test metadata:")
             self.log.debug("  filename: %s", self.filename)
+        try:
+            teststmpdir = self.teststmpdir
+        except EnvironmentError:
+            pass
+        else:
+            self.log.debug("  teststmpdir: %s", self.teststmpdir)
+        self.log.debug("  workdir: %s", self.workdir)
```

Comment:
Yes, I think we reached point where we do want to log those information, anyway such change should also change the `teststmpdir` and `workdir` from lazy properties to simple variables as they will always be evaluated.

Reply:
The problem with making them simple attributes, and not properties, is that overwriting them would be possible from a test.  Changes such as 9c35141c9e1441fc82300407a6cc614fa1aa79a0 took that into account.

Reply:
Sure, that's why I mentioned keeping them as properties, but set them on init rather than lazily evaluate them as they will always be set at least once.